### PR TITLE
Add esc 50 dataset a sound classification example

### PR DESCRIPTION
**Pull request by arthurPignet**
Add the [ESC-50: Dataset for Environmental Sound Classification](https://github.com/karolpiczak/ESC-50)

### CODE REVIEW

**Comment by bowni** [comment]
@arthurPignet a small detail: start your commit messages with a capital letter 😃

File: .gitignore [hunk 1]
```diff
@@ -110,3 +113,4 @@ nohup.out
 
 trash/
 experiments/
+datasets/local_data/
```
Comment (bowni) [inline .gitignore:116]: Adding a newline at the end of the file would be cleaner I think

File: config.yml [hunk 2]
```diff
@@ -1,59 +1,18 @@
-experiment_name: my_example
-n_repeats: 5
+experiment_name: debug_config
+n_repeats: 1
```
Comment (bowni) [inline config.yml:2]: OK, sorry it is unclear: try not to play with `config.yml` with has to stay a stable default config file for a basic experimentation. You can use `config_quick_debug.yml`, although it is also constrained by being simple enough so that the automated CI tests don't take too long. Actually we should clarify all that with a `default_config.yml`, a `ci_tests_config.yml` and a `dowhatyouwant_config.yml`

File: datasets/dataset_cifar10.py [hunk 3]
```diff
@@ -21,30 +22,47 @@ def preprocess_dataset_inputs(x):
...
+        y_test,
+        input_shape,
+        num_classes,
+        preprocess_dataset_labels,
+        generate_new_model_for_dataset,
+    )
+    return dataset_obj
+
+
+# Load and flatten data
```
Comment (bowni) [inline datasets/dataset_cifar10.py:58]: It seems these lines should go

File: datasets/dataset_mnist.py [hunk 4]
```diff
@@ -10,14 +10,41 @@
...
+        x_test,
+        y_train,
+        y_test,
+        input_shape,
+        num_classes,
+        preprocess_dataset_labels,
+        generate_new_model_for_dataset,
+    )
+    return dataset_obj
+
```
Comment (bowni) [inline datasets/dataset_mnist.py:42]: OK, cool

File: datasets/dataset_titanic.py [hunk 5]
```diff
@@ -75,11 +99,8 @@ def generate_new_model_for_dataset():
     model.add(Dense(8, activation='relu'))
     model.add(Dense(1, activation='sigmoid'))
 
-    model.compile(optimizer = 'adam',
-                   loss = 'binary_crossentropy',
-                   metrics = ['accuracy'])
+    model.compile(optimizer='adam',
+                  loss='binary_crossentropy',
+                  metrics=['accuracy'])
```
Comment (bowni) [inline datasets/dataset_titanic.py:104]: Way cleaner, great job

File: requirements.txt [hunk 6]
```diff
@@ -1,3 +1,4 @@
+librosa
```
Comment (bowni) [inline requirements.txt:1]: No version number?

File: scenario.py [hunk 7]
```diff
@@ -70,17 +72,7 @@ def __init__(self, params, experiment_path, scenario_id=1, n_repeat=1, is_dry_ru
...
-            dataset_module.x_train,
-            dataset_module.x_test,
-            dataset_module.y_train,
-            dataset_module.y_test,
-            dataset_module.input_shape,
-            dataset_module.num_classes,
-            dataset_module.preprocess_dataset_labels,
-            dataset_module.generate_new_model_for_dataset,
-        )
+        self.dataset = dataset_module.generate_new_dataset()
```
Comment (bowni) [inline scenario.py:75]: 👍

**Review by bowni — COMMENTED** [review]
Ok, haven't tested it yet but it looks great. See @arthurPignet a few comments on-the-go below:

File: main.py [hunk 8]
```diff
@@ -22,7 +22,7 @@
 import scenario
 import utils
 
-DEFAULT_CONFIG_FILE = "config.yml"
+DEFAULT_CONFIG_FILE = "config_quick_debug.yml"
```
Comment (RomainGoussault) [inline main.py:25]: Why ?

File: scenario.py [hunk 9]
```diff
@@ -43,7 +42,7 @@ def __init__(self, params, experiment_path, scenario_id=1, n_repeat=1, is_dry_ru
             raise Exception(f"Unrecognised parameters, check your .yml file")
 
         # Get and verify which dataset is configured
-        supported_datasets_names = ["mnist", "cifar10", "titanic"]
+        supported_datasets_names = ["mnist", "cifar10", "titanic", "esc50"]
```
Comment (RomainGoussault) [inline scenario.py:45]: We could add each dataset name to the constants.py file

File: datasets/dataset_esc50.py [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+    os.unlink(f'{path}/ESC-50.zip')
+    for src in glob.glob(f'{path}/ESC-50-master/audio'):
+        shutil.move(src, f'{path}/{os.path.basename(src)}')
+    shutil.move(f'{path}/ESC-50-master/meta/esc50.csv', f'{path}/esc50.csv')
+
+    shutil.rmtree(f'{path}/ESC-50-master')
+
+
+# Model structure and generation
+def generate_new_model_for_dataset():
```
Comment (RomainGoussault) [inline datasets/dataset_esc50.py:130]: Do you have reference for this model (a blog post / a paper) or did you build it on your own ?

File: datasets/dataset_esc50.py [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+def load_data():
+    """
+    load the dataset. Note that the x are iterators which need to be preprocess
+    :return: (x_train, y_train), (x_test, y_test)
+    """
+    path = Path(__file__).resolve().parents[0]
+    repertoire = str(path) + '/local_data/esc50'
+    if not os.path.exists(repertoire):
+        os.makedirs(repertoire)
+        os.chdir(repertoire)
```
Comment (RomainGoussault) [inline datasets/dataset_esc50.py:90]: [Pathlib ](https://docs.python.org/fr/3/library/pathlib.html)is a great alternative to `os.path` !

**Review by RomainGoussault — CHANGES_REQUESTED** [review]
Great work @arthurPignet, it's super clean ! 👍

File: datasets/dataset_esc50.py (hunk 10 above)
Comment (arthurPignet) [reply datasets/dataset_esc50.py:130]: Yes I have ! I add it

**Review by RomainGoussault — COMMENTED** [review]
Nice work @arthurPignet ! Two questions:

- What perfomance did you obtain with multiple partners ? 
- Can you write an end to end test or do you think it will take too long to run (i.e. more than 5 minutes) ?

File: constants.py [hunk 12]
```diff
@@ -21,3 +21,12 @@
 
 # Contributivity methods names
 # TODO
+
+# Datasets' Tags
+MNIST = "mnist"
+CIFAR10 = "cifar10"
+TITANIC = "titanic"
+ESC50 = "esc50"
```
[the remark this answers is not in the document]
Comment (arthurPignet) [reply constants.py:29]: With only one partner I reached 50% of accuracy (50 epochs), which is really good, considering the amount of classes and the few data. With two partners I obtained 44 %, with an early stop at 38 epochs (seq-pure)

When I trained the model direcly in tensorflow, even with 100 epochs and 64 samples per batch I never had more than 55-60% accuracy

Get higher accuracy than 60% would requiere data augmentation/deeper and more complexe networks and architecture

Note that I removed the local test/val splits of the data. (Well, not removed, but now these splits are datasets-dependant) 
I found out that the final train_set was composed only with 55% of the data, which was not enought for esc50 (the whole dataset contains 2000 samples)

**Review by bowni — APPROVED** [review]
Great work @arthurPignet , both on ESC50 and the associated ML problem, and on refactoring part of the dataset management!

**Review by RomainGoussault — APPROVED** [review]
Looks good to me ! Well done @arthurPignet  👍
